icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -51,35 +71,11 @@ private static String getWithPlural(String[] strings, StandardPlural plural) {
         return result;
     }
 
-    //////////////////////////
-    /// BEGIN DATA LOADING ///
-    //////////////////////////
-
-    private static final class PluralTableSink extends UResource.Sink {
-
-        String[] outArray;
-
-        public PluralTableSink(String[] outArray) {
-            this.outArray = outArray;
-        }
-
-        @Override
-        public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
-            UResource.Table pluralsTable = value.getTable();
-            for (int i = 0; pluralsTable.getKeyAndValue(i, key, value); ++i) {
-                int index = getIndex(key.toString());
-                if (outArray[index] != null) {
-                    continue;
-                }
-                String formatString = value.getString();
-                outArray[index] = formatString;
-            }
-        }
-    }
-
-    // NOTE: outArray MUST have at least ARRAY_LENGTH entries. No bounds checking is performed.
+    ////////////////////////
+    /// END DATA LOADING ///
```

Comment:
BEGIN and END DATA LOADING used to be around resource sinks and getMeasureData. Now they're around the constructor, getIndex, and getWithPlural - I don't think this is a correct change. Please undo?

Reply:
good catch, we can move it or return the code to its original place.

Reply:
done

Reply:
Nope: PluralTableSink is still at the bottom of the file, not between BEGIN and END.

Reply:
And getUnitDisplayName is inside BEGIN and END DATA LOADING, when it doesn't load data (it does call getMeasureData, but all functions calling getMeasureData are outside this BEGIN/END).

I'm moving these blocks of code back to where they were.

Reply:
Fixed in 08744e0867